### PR TITLE
Allow FP16 focal loss compute and fix a few comments.

### PR DESCRIPTION
**Pull request by mingxingtan**

### CODE REVIEW

**Review by fsx950223 — APPROVED** [review]

**Review by NikZak — APPROVED** [review]

File: efficientdet/det_model_fn.py [hunk 1]
```diff
@@ -613,15 +612,15 @@ def scaffold_fn():
...
+    class OomReportingHook(tf.estimator.SessionRunHook):
 
-        def before_run(self, run_context):
-          return tf.estimator.SessionRunArgs(
-              fetches=[],
-              options=tf.RunOptions(report_tensor_allocations_upon_oom=True))
+      def before_run(self, run_context):
+        return tf.estimator.SessionRunArgs(
+            fetches=[],
+            options=tf.RunOptions(report_tensor_allocations_upon_oom=True))
```
Comment (fsx950223) [inline efficientdet/det_model_fn.py:621]: Does oom  tracker will decrease training speed?

Comment (mingxingtan) [reply efficientdet/det_model_fn.py:621]: It shouldn't, as it  is triggered only if OOM happens. Did you have this experience before?

Comment (fsx950223) [reply efficientdet/det_model_fn.py:621]: I speed up training process by remove profilehook and oomhook.
I'm not care about each hook's performance.

Comment (mingxingtan) [reply efficientdet/det_model_fn.py]: You are right, this hook indeed slow down the training, I have disabled it in default.

**Review by fsx950223 — APPROVED** [review]

**Pull request merged**